docs/sources/flow/reference/components/loki.server.md

```diff
@@ -0,0 +1,86 @@
+---
+title: loki.server
+---
+
+# loki.server
+
+`loki.server` is a common block used in all `loki.` components that expose a network server, such as
+[loki.source.gcplog][] and [loki.source.heroku][]. The
+network server supports both `HTTP` and `gRPC` protocols.
+
+[loki.source.gcplog]: {{< relref "./loki.source.gcplog.md" >}}
+[loki.source.heroku]: {{< relref "./loki.source.heroku.md" >}}
+
+## Usage
+
+```river
+server {
+    http {
+        listen_port = 8080
+        listen_adress = "0.0.0.0"
+    }
+    grpc {
+        listen_port = 8081
+    }
+}
+```
+
+## Arguments
+
+`loki.server` supports the following arguments:
+
+ Name                        | Type           | Description                               | Default | Required 
+-----------------------------|----------------|-------------------------------------------|---------|----------
+ `graceful_shutdown_timeout` | `RelabelRules` | Relabeling rules to apply on log entries. | "{}"    | no       
+
+## Blocks
+
+The following blocks are supported inside the definition of
+`loki.source.gcplog`:
+
+ Hierarchy | Name     | Description                 | Required 
+-----------|----------|-----------------------------|----------
+ http      | [http][] | Configures the HTTP server. | no       
+ push      | [grpc][] | Configures the gRPC server. | no       
+
+The `pull` and `push` inner blocks are mutually exclusive; a component must
+contain exactly one of the two in its definition.
+
+[http]: #http-block
+
+[grpc]: #grpc-block
+
+### http block
+
+The `http` configures the HTTP server.
+
+The following arguments can be used to configure the `http` block. Any omitted
+fields take their default values.
+
+ Name                   | Type       | Description                                                                                                          | Default  | Required 
+------------------------|------------|----------------------------------------------------------------------------------------------------------------------|----------|----------
+ `listen_address`       | `string`   | Network address on which the server will listen for new connections. Defaults to accepting all incoming connections. | `""`     | no       
+ `listen_port`          | `int`      | Port number on which the server will listen for new connections.                                                     | `9090`   | no       
```

Comment:
By default, Prometheus listens to :9090.

I'd say we change this and the gRPC default port to 8080/8081 which should be a tad safer.